uswds/uswds
# USWDS - Button: Create story for links styled as buttons

# Summary

Created a storybook preview for links styled as buttons. The story matches existing button storybook previews.

## Breaking change

This is not a breaking change.

## Related issue

Closes #5254 

## Related pull requests

~[Changelog →](https://github.com/uswds/uswds-site/pull/2242)~

This is an internal change.

## Preview link

[Links styled as buttons →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-button-link-story/?path=/story/components-button--test-links-as-buttons)

## Problem statement

Currently the only links with the `usa-button` class in storybook are the "back" button on [button-group](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-button-group--default) & the [card buttons](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/cm-button-group-a11y/iframe.html?args=&id=components-card--default) (after [#4385](https://github.com/uswds/uswds/pull/4385))

## Solution

Create a storybook preview for links styled as buttons.

### Notes:

- Removed `disabled` button from link preview
    - Anchor elements cannot be disabled via the attribute.
    - Users must use our disabled classes + remove pointer events to disable links
- Excluded `type` storybook control from Link story
    - I excluded the `type` attribute from the Links as Buttons template as well as the control. Anchor elements do not have `type` functionality.
- `aria_disabled` control
    - Control appears to be outdated and is not mentioned in any button templates. Submitted a ticket to remove. (#5436) 
    

## Testing and review

1. Inspect storybook preview
2. Confirm controls work
3. Confirm storybook includes all necessary elements to showcase links styled as buttons

### Testing Checklist

- [ ] Story appropriately showcases links styled as buttons
- [ ] Story and controls are configurated as we would prefer
- [ ] Test template matches existing patterns
- [ ] Variants apply correctly

## Further Considerations

Should we add links styled as buttons to the component preview?

<details>
<summary>Image</summary>

![image](https://github.com/uswds/uswds/assets/25211650/67d1c9dc-ab24-43b4-87f2-3f38336bd6dd)
</details>
